VDA-XP-Reward/contracts/VDAXPReward.sol

```diff
@@ -0,0 +1,277 @@
+
+    function __VDAXPReward_init_unchained(IERC20Upgradeable token, IVeridaDIDRegistry didRegistry) internal {
+        rewardToken = token;
+        didRegistryContract = didRegistry;
+        rateDenominator = 10000000; // Set up rate from 0.000001
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function getTokenAddress() external virtual view override returns(address) {
+        return address(rewardToken);
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function getDIDRegistryAddress() external virtual view override returns(address) {
+        return address(didRegistryContract);
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function getRateDenominator() external virtual view override returns(uint32) {
+        return rateDenominator;
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function setRateDenominator(uint32 denominator) external virtual override onlyOwner {
+        if (rateDenominator == denominator || denominator == 0) {
+            revert InvalidValue();
+        }
+        uint32 orgVal = rateDenominator;
+        rateDenominator = denominator;
+        emit UpdateRateDenominator(orgVal, denominator);
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function getConversionRate() external virtual view override returns(uint32) {
+        return conversionRate;
+    }
+
+    /**
+     * @dev See {IVDAXPReward}
+     */
+    function setConversionRate(uint32 newRate) external virtual override onlyOwner {
+        if (newRate == 0 || newRate == conversionRate) {
+            revert InvalidValue();
+        }
+        uint32 orgVal = conversionRate;
+        conversionRate = newRate;
+        emit UpdateConversionRate(orgVal, newRate);
+    }
+
+    /**
+     * @notice Validate the time of proof
+     * @dev Separated for `stack too deep`
+     * @param didAddress - DID address. Used to revert
+     * @param curTime - Current time
+     * @param info - Claim information
+     */
+    function _validateProofTime(address didAddress, DateTime._DateTime memory curTime, ClaimInfo calldata info) internal virtual view {
+        if (curTime.year != info.issueYear || curTime.month != (info.issueMonth + 1)) {
+            revert InvalidProofTime(didAddress, info.typeId, info.issueYear, info.issueMonth);
+        } else if (curTime.month == 1) {
+            if (curTime.year != (info.issueYear + 1) || info.issueMonth != 12) {
+                revert InvalidProofTime(didAddress, info.typeId, info.issueYear, info.issueMonth);
+            }
+        }
+    }
+
+    /**
+     * @notice Validate proof signer
+     * @dev Separated for `stack too deep`
+     * @param didAddress - DID address, Used to revert
+     * @param proofSigner - Signer of proof
+     * @param info - Claim information
+     */
+    function _validateProofSigner(address didAddress, address proofSigner, ClaimInfo calldata info) internal virtual view {
+        bool isVerified;
+        uint index;
```

Comment:
Does solidity default value - `0` for index?

Reply:
Correct